soc/xtensa/esp32/soc.h

```diff
@@ -31,6 +31,16 @@ static inline void esp32_clear_mask32(uint32_t v, uint32_t mem_addr)
 	sys_write32(sys_read32(mem_addr) & ~v, mem_addr);
 }
 
+static inline uint32_t esp_core_id(void)
```

Comment:
cant you use existing `arch_proc_id()`?

Reply:
Not possible yet, arch_proc_id() was developed with SMP in mind, that is it, it relies the on the SMP code to initialize the proper CPU pointers during kernel initialization. In AMP scenarios this code is not linked, and it would need to be modified.

Using it without SMP code makes the arch_proc_id() to return always CPU0

Reply:
@nashif , what do you think?